src/seedu/addressbook/AddressBook.java

```diff
@@ -411,7 +436,59 @@ private static String getMessageForSuccessfulAddPerson(String[] addedPerson) {
         return String.format(MESSAGE_ADDED,
                 getNameFromPerson(addedPerson), getPhoneFromPerson(addedPerson), getEmailFromPerson(addedPerson));
     }
+    
+    /**
+     * Update T2A5:
+     * Edits an existing person (specified by the command args) in the address book.
+     * The entire command arguments string is treated as a string representation of the person to edit.
+     *
+     * @param commandArgs full command args string from the user
+     * @return feedback display message for the operation result
+     */
+    private static String executeEditPerson(String commandArgs) {
+        // try decoding a person from the raw args
+        final Optional<String[]> decodeResult = decodePersonFromString(commandArgs);
+
+        // checks if args are valid (decode result will not be present if the person is invalid)
+        if (!decodeResult.isPresent()) {
+            return getMessageForInvalidCommandInput(COMMAND_EDIT_WORD, getUsageInfoForAddCommand());
+        }
 
+        // add the person as specified
+        final String[] personToEdit = decodeResult.get();
+        if (editPersonInAddressBook(personToEdit)) {
+        	return getMessageForSuccessfulEditPerson(personToEdit);
+    	} else {
+    		return getMessageForFailedEditPerson(personToEdit);
+    	}
+    }
+    
+    /**
+     * Update T2A5:
```

Comment:
As mentioned during the tutorial, avoid comments like this which are meant for the reviewer, not future readers of the code.